python/pecos/simulators/parent_sim_classes.py

```diff
@@ -34,13 +34,16 @@ def run_gate(self, symbol: str, locations: set[int] | set[tuple[int, ...]], **pa
         output = {}
 
         if params.get("simulate_gate", True):
-            for location in locations:
-                results = self.bindings[symbol](self, location, **params)
+            if locations:
```

Comment:
This code block may require careful changes. I ran into a problem where two-qubit gates such as `RZZ` are being passed their arguments one by one (`location` here) and, hence failing.

See
```py
RZZ, [19, 0], None, {'angles': [0.0]}, None
{'angles': [0.0], 'angle': 0.0}
19
Traceback (most recent call last):
  File "/Users/kartik.singhal/GH/pytket-phir/./pc.py", line 329, in <module>
    main()
  File "/Users/kartik.singhal/GH/pytket-phir/./pc.py", line 300, in main
    print(HybridEngine(qsim="state-vector").run(program=p, shots=10))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/kartik.singhal/GH/PECOS/python/pecos/engines/hybrid_engine.py", line 168, in run
    measurements = self.qsim.run(noisy_buffered_qops)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/kartik.singhal/GH/PECOS/python/pecos/simulators/quantum_simulator.py", line 67, in run
    output = self.state.run_gate(op.name, op.args, **op.metadata)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/kartik.singhal/GH/PECOS/python/pecos/simulators/parent_sim_classes.py", line 42, in run_gate
    results = self.bindings[symbol](self, location, **params)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/kartik.singhal/GH/PECOS/python/pecos/simulators/projectq/gates_two_qubit.py", line 161, in RZZ
    q1 = state.qids[qubits[0]]
                    ~~~~~~^^^
TypeError: 'int' object is not subscriptable
```